src/constants/routes.ts

```diff
@@ -1,7 +1,29 @@
-export const ROUTES = {
-  home: '/',
-  signIn: '/login',
-  signUp: '/register',
-  articleDetail: '/detail/',
-  userArticlePage: '/user/',
-};
+interface Route {
+  name: string;
+  href: string;
+  nav: boolean;
+}
+
+interface Routes {
+  HOME: Route;
+  SIGNIN: Route;
+  SIGNUP: Route;
+  ARTICLEDETAIL: (slug: string) => Route;
+  USERARTICLEPAGE: (userId: string) => Route;
```

Comment:
단어 단위로 언더바로 끊어주면 좋습니다~

```suggestion
  ARTICLE_DETAIL: (slug: string) => Route;
  USER_ARTICLE_PAGE: (userId: string) => Route;
```